### PR TITLE
scenario xml definition

### PR DESCRIPTION
**Pull request by AcidSepp**

### CODE REVIEW

File: LaTeX/Parts/Chapter4.tex [hunk 1]
```diff
@@ -29,6 +29,9 @@ \chapter{Scenario Concept}
 </scenario>
 \end{lstlisting}
 
+A Varroa test is the execution of a user defined Scenario, that is marked down in a XML file.
+To define a valid scenario the user needs to define the topology of the Scenario and the behaviour of its components.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter4.tex:33]: define occures twice, maybe use synonym e.g. characterize / describe once

File: LaTeX/Parts/Chapter4.tex [hunk 2]
```diff
@@ -38,6 +41,13 @@ \section{Broker}
 	</broker>
 </broker>
 \end{lstlisting}
+The broker is the central component whose resilience is tested by the Scenario.
+The user needs to define:
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter4.tex]: Maybe start the sentence with 'Herefore', could be more fluently readable.

File: LaTeX/Parts/Chapter4.tex [hunk 3]
```diff
@@ -47,13 +57,22 @@ \section{Client Groups}
...
 			</mqtt>
 		</clientGroup>
 
 		<!-- definiton of more client groups -->
 </scenario>
 \end{lstlisting}
+A Scenario contains a number of Client Groups.
+A Client Group is a specific amount of MQTT Clients that share the exact same behaviour.
+When defining a client group the user needs to specify:
+\begin{itemize}
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter4.tex:70]: Put full stops at the end of each sentence used in the itemize.

File: LaTeX/Parts/Chapter4.tex [hunk 4]
```diff
@@ -68,6 +87,14 @@ \section{Topic Groups}
 	</topicGroups>
 </scenario>
 \end{lstlisting}
+A Scenario contains a number of Topic Groups.
+A Topic Group is a specific amount of MQTT Topics whose names are created from the same regular expression.
+When defining a Topic Group the user needs to specify:
+\begin{itemize}
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter4.tex:93]: Put full stops at the end of each sentence used in the itemize.

File: LaTeX/Parts/Chapter4.tex [hunk 5]
```diff
@@ -78,19 +105,36 @@ \section{Subscriptions}
...
+		<subscription id="sub-2">
+			<topicFilter>/topic/subtopic/subsubtopic/#</topicFilter>
+		</subscription>
+
 		<!-- definition of more subscriptions -->
 	</subscriptions>
 </scenario>
 \end{lstlisting}
+A Scenario can contain Subscriptions.
+A Subscription defines a subscription behaviour that Client Groups can implement in their Subscribe Commands.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter4.tex]:
```suggestion
A Subscription defines a certain subscription behaviour that Client Groups can implement in their Subscribe Commands.
```

File: LaTeX/Parts/Chapter4.tex [hunk 6]
```diff
@@ -100,5 +144,55 @@ \section{Stages}
 	</stages>
 </scenario>
 \end{lstlisting}
+A Scenario is divided into Stages.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter4.tex]:
```suggestion
A Scenario is divided into one or more Stages.
```

File: LaTeX/Parts/Chapter4.tex [hunk 7]
```diff
@@ -29,6 +29,9 @@ \chapter{Scenario Concept}
 </scenario>
 \end{lstlisting}
 
+A Varroa test is the execution of a user defined Scenario, that is marked down in a XML file.
```
Comment (sauroter) [inline LaTeX/Parts/Chapter4.tex]:
```suggestion
A Varroa test is the execution of a user defined Scenario, which is specified in a XML file.
```

File: LaTeX/Parts/Chapter4.tex [hunk 8]
```diff
@@ -100,5 +144,55 @@ \section{Stages}
 	</stages>
 </scenario>
 \end{lstlisting}
+A Scenario is divided into Stages.
+Those Stages are executed sequential in the order they are specified in the XML document.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter4.tex]:
```suggestion
These Stages are executed in sequential order as they are specified in the XML document.
```

File: LaTeX/Parts/Chapter4.tex [hunk 9]
```diff
@@ -100,5 +144,55 @@ \section{Stages}
...
+A Stage contains a number of Lifecycles, those specify the behaviour of Client Groups in the stage.
+Lifecylces within a stage are executed in parallel.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter4.tex]:
```suggestion
Lifecycles within a stage are executed in parallel.
```

File: LaTeX/Parts/Chapter4.tex [hunk 10]
```diff
@@ -38,6 +41,13 @@ \section{Broker}
 	</broker>
 </broker>
 \end{lstlisting}
+The broker is the central component whose resilience is tested by the Scenario.
```
Comment (sauroter) [inline LaTeX/Parts/Chapter4.tex]:
```suggestion
The broker is the central component whose performance and stress resistance is tested by the Scenario.
```

File: LaTeX/Parts/Chapter4.tex [hunk 11]
```diff
@@ -100,5 +144,55 @@ \section{Stages}
 	</stages>
 </scenario>
 \end{lstlisting}
+A Scenario is divided into Stages.
+Those Stages are executed sequential in the order they are specified in the XML document.
+A Stage contains a number of Lifecycles, those specify the behaviour of Client Groups in the stage.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter4.tex]:
```suggestion
A Stage contains a number of Lifecycles, which specify the behaviour of Client Groups in the stage.
```

File: LaTeX/Parts/Chapter4.tex [hunk 12]
```diff
@@ -100,5 +144,55 @@ \section{Stages}
...
+Lifecylces within a stage are executed in parallel.
+To define a stage the user needs to specify:
+\begin{itemize}
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter4.tex:152]: Add full stops at the end of items.

File: LaTeX/Parts/Chapter4.tex [hunk 13]
```diff
@@ -78,19 +105,36 @@ \section{Subscriptions}
...
+To define a Topic Group the user needs to specify:
+\begin{itemize}
+	\item \textbf{id:} the identifier to reference the Subscription
+	\item \textbf{topicFilter:} the Topic Filter to target specific Topics
+\end{itemize}
+or
+\begin{itemize}
+	\item \textbf{id:} the identifier to reference the Subscription
+	\item \textbf{topicGroup:} reference to a Topic Group
+	\item \textbf{wildCard:} whether the Subscription is wild-carded
```
Comment (sauroter) [inline LaTeX/Parts/Chapter4.tex]: I don't think wild-carded is a word

File: LaTeX/Parts/Chapter4.tex (hunk 1 above)
Comment (AcidSepp) [reply LaTeX/Parts/Chapter4.tex:33]: no

File: LaTeX/Parts/Chapter4.tex (hunk 2 above)
Comment (AcidSepp) [reply LaTeX/Parts/Chapter4.tex]: 'For this' is better, but good idea.

**Pull request merged**